### PR TITLE
Get xrefs from thoas

### PR DESCRIPTION
**Pull request by msmanoj**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-622

## Description
External references displayed in entity view right hand panel are now retrieved from Thoas endpoint.

## Views affected
Entity viewer -> RH panel -> External references

http://hx-rke-wp-webadmin-14-worker-1.caas.ebi.ac.uk:32756/app/entity-viewer/homo_sapiens_GCA_000001405_27/homo_sapiens_GCA_000001405_27:gene:ENSG00000139618

### Other effects
- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

### CODE REVIEW

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/GeneViewSideBar.tsx [hunk 1]
```diff
@@ -13,14 +15,19 @@ type Props = {
...
 };
 
-const GeneViewSidebar = (props: Props) => {
-  if (props.activeTabName === SidebarTabName.OVERVIEW) {
-    return <GeneOverview />;
-  } else if (props.activeTabName === SidebarTabName.EXTERNAL_REFERENCES) {
-    return <GeneExternalReferences />;
-  }
+const client = new ApolloClient({
+  uri: '/thoas'
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/GeneViewSideBar.tsx:19]: Ah! This is a good reminder that Thoas does not support cross-origin requests, which it definitely should.

Comment (azangru) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/GeneViewSideBar.tsx:19]: Submitted a PR for Thoas to enable support of cross-origin requests: https://github.com/Ensembl/ensembl-thoas/pull/8. We'll have to wait for Kieron to accept it until the feature branch can properly work. The `/thoas` endpoint with the corresponding proxy config in the dev server will then become redundant.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import React from 'react';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:1]: Reading through the code of this file, I got confused by at least three different names used in this file for a very similar concept:

1) CrossReference
2) ExternalLinks
3) Xrefs

I wonder whether using two variables clearly showing their relationships to each other, e.g. `crossReference` and `crossReferenceGroup`, where crossReferenceGroup would have the type of:

```
{
  source: Source;
  references: CrossReferenceWithoutSource
}
```
would not make the code clearer.

Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:17]: Thanks. That's done now.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/GeneViewSideBar.tsx (hunk 1 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/GeneViewSideBar.tsx]: I've now removed the proxy config. Thanks.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx [hunk 3]
```diff
@@ -99,16 +190,19 @@ const GeneExternalReferences = (props: Props) => {
   );
 };
 
-const renderXrefGroupWithSameLabels = (xref: ExternalLink, key: number) => {
+const renderXrefGroupWithSameLabels = (
+  xref: CrossReferenceGroup,
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:194]: `xref ` -> `group`?

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx [hunk 4]
```diff
@@ -29,62 +32,150 @@ import ExternalReference from 'src/shared/components/external-reference/External
...
+  transcripts: {
+    stable_id: string;
+    cross_references: CrossReference[];
+  }[];
+  cross_references: CrossReference[];
 };
 
-const GeneExternalReferences = (props: Props) => {
-  if (!props.sidebarPayload) {
+const getFormattedCrossReferences = (crossReferences: CrossReference[]) => {
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:87]: what does "formatted cross-references" mean?

is this function just building cross-reference groups? should it be called `buildCrossReferenceGroups` or `groupCrossReferences`?

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx [hunk 5]
```diff
@@ -29,62 +32,150 @@ import ExternalReference from 'src/shared/components/external-reference/External
...
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!data || !data.gene) {
     return <div>No data to display</div>;
   }
+
+  const geneCrossReferences = getFormattedCrossReferences(
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:133]: `geneCrossReferences` -> `crossReferenceGroups`?

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx [hunk 6]
```diff
@@ -29,62 +32,150 @@ import ExternalReference from 'src/shared/components/external-reference/External
...
+        <div className={styles.geneSymbol}>{data.gene.name}</div>
+        <div className={styles.stableId}>{data.gene.stable_id}</div>
       </div>
 
       <div className={styles.sectionHead}>Gene</div>
-      {props.sidebarPayload.gene?.xrefs &&
-        props.sidebarPayload.gene.xrefs.map((xref, key) => {
-          if (xref.links.length === 1) {
+      {data.gene.cross_references &&
+        Object.values(geneCrossReferences).map((xrefs, key) => {
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx:147]: `xrefs` -> `group`?

Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx]: That's done

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx (hunk 5 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx]: That's done

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx (hunk 4 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx]: Renamed it to `buildCrossReferenceGroups`

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx (hunk 3 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx]: Renamed it to `crossRefereneGroup`

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx (hunk 6 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx]: Renamed it to `crossRefereneGroup`

**Pull request merged**